src/components/Datepicker/Datepicker.PeriodCalendar.js

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { noop } from '../../utilities/other'
+import { classNames } from '../../utilities/classNames'
+import { getActiveYearRange } from './Datepicker.utils'
+import Navigator from './Datepicker.Navigator'
+import { PeriodUI, PeriodButtonUI } from './Datepicker.css'
+import { MONTHS, NAVIGATION_LEVELS } from './Datepicker.constants'
+
+function PeriodCalendar({
+  activeMonths,
+  allowFutureDatePick,
+  goToDate,
+  goToPreviousYear,
+  goToNextYear,
+  startDate,
+}) {
+  const todaysDate = new Date()
+  const todaysMonth = todaysDate.getMonth()
+  const todaysYear = todaysDate.getFullYear()
+  const activeMonth = activeMonths[0].month
+  const activeYear = activeMonths[0].year
+  const selectedYear = startDate ? startDate.getFullYear() : ''
+  const [mode, setMode] = useState(NAVIGATION_LEVELS.YEAR_BY_YEAR)
+  const [yearRange, setYearRange] = useState(
+    getActiveYearRange(activeYear, true)
+  )
+  const firstYearInRange = yearRange[0]
+  const lastYearInRange = yearRange[yearRange.length - 1]
+  const navigatorLabel =
+    mode === NAVIGATION_LEVELS.YEAR_BY_YEAR
+      ? activeYear
+      : `${firstYearInRange}–${lastYearInRange}` // note the 'en dash'
+
+  function canNavigateForward() {
+    if (mode === NAVIGATION_LEVELS.YEAR_BY_YEAR) {
+      return allowFutureDatePick || activeYear < todaysYear
+    }
+
+    return allowFutureDatePick || lastYearInRange < todaysYear
+  }
+
+  return (
+    <div className="c-PeriodCalendar">
+      <Navigator
+        canNavigateForward={canNavigateForward()}
+        navigationLevel={mode}
+        label={navigatorLabel}
+        goNext={() => {
+          if (mode === NAVIGATION_LEVELS.YEAR_BY_YEAR) {
+            goToNextYear()
+          } else {
+            setYearRange(getActiveYearRange(lastYearInRange + 13, false))
+          }
+        }}
+        goPrevious={() => {
+          if (mode === NAVIGATION_LEVELS.YEAR_BY_YEAR) {
+            goToPreviousYear()
+          } else {
+            setYearRange(getActiveYearRange(firstYearInRange, false))
+          }
+        }}
+        onDeepNavigationClick={() => {
+          if (mode === NAVIGATION_LEVELS.YEAR_BY_YEAR) {
+            setMode(NAVIGATION_LEVELS.YEAR_RANGES)
+          }
+        }}
+      />
+      <PeriodUI
+        role="grid"
+        className={classNames(
+          'c-Period',
+          mode === NAVIGATION_LEVELS.YEAR_BY_YEAR
+            ? 'is-mode-months'
+            : 'is-mode-years'
+        )}
+      >
+        {mode === NAVIGATION_LEVELS.YEAR_BY_YEAR
```

Comment:
the ternary condition is a bit confusing here, just because it's on multiple line. maybe we could extract that to a function to render the right content?

Reply:
No problem, done!